packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -333,6 +424,429 @@ private MethodProvider BuildEmptyConstructor()
+        /// If the statement is not an IfStatement, we just create an IfStatement and return.
+        /// If the statement is an IfStatement, we could add the condition to its condition which should simplify the generated code.
+        /// </summary>
+        private IfStatement WrapInCheckNotWireIfStatement(ValueExpression format, MethodBodyStatement statement)
+        {
+            var isNotWireCondition = NotEqual(format, ModelReaderWriterOptionsSnippet.WireFormat);
+            if (statement is IfStatement ifStatement)
+            {
+                var updatedCondition = And(isNotWireCondition, new BoolSnippet(ifStatement.Condition));
+                IfStatement updatedIf = new(updatedCondition, ifStatement.Inline, ifStatement.AddBraces)
+                {
+                    ifStatement.Body
+                };
+                return updatedIf;
+            }
+
+            return new IfStatement(isNotWireCondition)
+            {
+                statement
+            };
+        }
+
+        /// <summary>
+        /// Creates a serialization statement for the specified type.
+        /// </summary>
+        /// <param name="serializationType">The type being serialized.</param>
+        /// <param name="value">The value to be serialized.</param>
+        /// <param name="serializationFormat">The serialization format.</param>
+        /// <returns>The serialization statement.</returns>
+        /// <exception cref="NotSupportedException">Thrown when the serialization type is not supported.</exception>
+        private MethodBodyStatement CreateSerializationStatement(
+            CSharpType serializationType,
+            ValueExpression value,
+            SerializationFormat serializationFormat)
+        {
+            MethodBodyStatement? serializationStatement = null;
+            switch (serializationType)
+            {
+                case var dictionaryType when dictionaryType.IsDictionary:
+                    serializationStatement = CreateDictionarySerializationStatement(
+                        new DictionarySnippet(dictionaryType.Arguments[0], dictionaryType.Arguments[1], value),
+                        serializationFormat);
+                    break;
+                case var listType when listType.IsList || listType.IsArray:
+                    serializationStatement = CreateListSerializationStatement(
+                        GetEnumerableExpression(value, listType),
+                        serializationFormat);
+                    break;
+                case var frameworkType when !frameworkType.IsCollection:
+                    serializationStatement = CreateValueSerializationStatement(serializationType, serializationFormat, value);
+                    break;
+            }
+
+            return serializationStatement ?? throw new NotSupportedException($"Serialization of type {serializationType.Name} is not supported.");
```

Comment:
can return from each case and throw in the default which will let you turn this method into a switch lambda which is more performant